### PR TITLE
Handle mongo's option explicitly

### PR DESCRIPTION
**Pull request by minkj1992**
- #38 
- https://www.popit.kr/mongodb-golang-%EB%93%9C%EB%9D%BC%EC%9D%B4%EB%B2%84%EC%9D%98-%EC%BB%A8%ED%85%8D%EC%8A%A4%ED%8A%B8%EC%99%80-%EC%BB%A4%EB%84%A5%EC%85%98/

### CODE REVIEW

File: infrastructure/persistence/db/mongo.go [hunk 1]
```diff
@@ -20,7 +21,22 @@ type MongoInstance struct {
 
 func NewMongoDB(config *configs.Config) (*MongoInstance, error) {
 	mongoURI := config.MongoURI + "/" + config.MongoDBName
-	client, err := mongo.NewClient(options.Client().ApplyURI(mongoURI))
+	clientOptions := options.Client().ApplyURI(mongoURI)
+
+	switch config.Phase {
+	case configs.Production:
+		clientOptions.SetHeartbeatInterval(15 * time.Second)
+		clientOptions.SetMaxPoolSize(100)
```
Comment (devjaewon) [inline infrastructure/persistence/db/mongo.go:29]: This is mere curiosity. What is the basis for determining the pool size?
experience?

**Review by devjaewon — APPROVED** [review]
good!!

File: infrastructure/persistence/db/mongo.go (hunk 1 above)
Comment (minkj1992) [reply infrastructure/persistence/db/mongo.go:29]: I don't know this part either.

@mcauto ,@jiwonkim-park 
Do you guys have any idea how to calculate the proper number of db connection pools?

Comment (mcauto) [reply infrastructure/persistence/db/mongo.go:29]: > I don't know this part either.
> 
> @mcauto ,@jiwonkim-park Do you guys have any idea how to calculate the proper number of db connection pools?

I recommend measuring and deciding the performance :)

> **"If you think it’s slow, prove it with a benchmark."**
> [the zen of go](https://dave.cheney.net/2020/02/23/the-zen-of-go)
> **“In the face of ambiguity, refuse the temptation to guess.”**
> The Zen of Python, Item 12

https://github.com/codesenberg/bombardier  
https://github.com/rakyll/hey

Comment (minkj1992) [reply infrastructure/persistence/db/mongo.go:29]: @mcauto Thank you for always sharing good opensources

Comment (mcauto) [reply infrastructure/persistence/db/mongo.go:29]: Thread pool appropriate size = cpu core count * (1 / cpu usage time).
> https://link.medium.com/4WqDHttHXrb

The cpu usage will be just an estimate, so it needs to be adjusted every time.
It seems necessary to make it easy to change at any time with environmental variables or setting values.

**Pull request merged**